stl_text/datamodule/doc_classification.py

```diff
@@ -27,17 +51,23 @@ def __init__(self, data_path: str = 'glue_sst2_tiny', vocab_path: Optional[str]
         self.text_transform = None
         self.label_transform = None
         self.datasets = {}
+        self.unk_token = '<unk>'
 
     def setup(self, stage):
-        # torchtext's spm_tokenizer and vocab are not pickleable and will fail dataset.map()
-        # tokenizer = spm_tokenizer(download_from_url(PRETRAINED_SP_MODEL["text_unigram_25000"]))
-        # vocab = build_fairseq_vocab("vocab_tiny.txt")
+        tokenizer = sentencepiece_tokenizer(download_from_url(PRETRAINED_SP_MODEL["text_unigram_25000"])).to_ivalue()
+        if self.vocab_path:
+            path_manager = PathManager()
+            with path_manager.open(self.vocab_path, "r",encoding='utf-8') as f:
+                vocab = build_vocab_from_text_file(f,torch.jit.script(tokenizer))
+        else:
+            vocab = Vocab(VocabPybind([self.unk_token],self.unk_token))
+
+        vocab = vocab.to_ivalue()
 
-        # sequential_transforms is not torchscriptable
-        # self.text_transform = sequential_transforms(tokenizer, vocab)
+        self.text_transform = TextToIndices(tokenizer,vocab)
 
-        # use dummy text_transform and label_transform to get rid of pickling and torchscript issues
-        self.text_transform = WhitespaceTokenizer(vocab_path=self.vocab_path)
+        # usage of below tokenizer is same as above except that basic_english_normalize() is used as tokenizer
+        # self.text_transform = WhitespaceTokenizer(vocab_path=self.vocab_path)
```

Comment:
We should find a way to support both tokenizers in the workflow. The ultimate solution will be `config`. For now, can we pass an argument?